sdv/utils.py

```diff
@@ -73,23 +73,34 @@ def get_datetime_format(value):
 def is_datetime_type(value):
     """Determine if the input is a datetime type or not.
 
+    If a ``pandas.Series`` or ``list`` is passed, it will return ``True`` if all the values
+    are datetime. Otherwise, it will check if the value is a datetime.
+
+    Note: it will return ``False`` if ``value`` is a string representing
+    a date before the year 1677.
+
     Args:
-        value (pandas.DataFrame, int, str or datetime):
+        value (array-like iterable, int, str or datetime):
             Input to evaluate.
 
     Returns:
         bool:
             True if the input is a datetime type, False if not.
     """
-    if isinstance(value, Iterable) and not isinstance(value, str):
-        value = get_first_non_nan_value(value)
```

Comment:
Why were we doing this before (selecting only the first value of a list)? Was it because of performance?

Reply:
Because it crashed at certain points on certain python versions (I believe it was 3.7). Pandas itself does this, gets first not `nan` value to find the `datetime` but for some reasons in python 3.7 it wasn't working as intended (it was failing with `None` or some other combinations).